File: data/TiendaMusical/Disco.java
import java.util.HashMap;

/**
 * Clase que representa un disco musical.
 */
public class Disco extends Producto {
    private String artista;
    private int anioLanzamiento;
    /**
     * Mapa que almacena las canciones del disco.
     * La clave es el número de pista (1, 2, 3, ...) y el valor es el nombre de la canción.
     */
    private HashMap<Integer, String> canciones; // Número de pista -> Nombre canción

    public Disco(String id, String titulo, double precio, String artista, int anioLanzamiento) {
        super(id, titulo, precio);
        this.artista = artista;
        this.anioLanzamiento = anioLanzamiento;
        this.canciones = new HashMap<>();
    }

    /**
     * Añade una canción al disco.
     * No debe hacer ninguna validacion extra.
     * @param numeroPista Número de pista
     * @param cancion Nombre de la canción
     */
    public void añadirCancion(int numeroPista, String cancion) {
        //TO DO: Implementar el método
    }

    /**
     * Calcula la duración total estimada del disco.
     * Se asume que cada canción dura aproximadamente 3.5 minutos.
     * @return Duración en minutos (estimación)
     */
    public double calcularDuracionTotal() {
        // TO DO
        return 0.0;
    }

    // Setters y Getters
    public String getArtista() {
        return artista;
    }
    public int getAñoLanzamiento() {
        return anioLanzamiento;
    }
    public HashMap<Integer, String> getCanciones() {
        return new HashMap<>(canciones);
    }
}

File: data/TiendaMusical/Dependencies/Disco.js
/**
 * Clase que representa un disco musical.
 * Asume que existe una clase Producto de la que hereda.
 */
class Disco extends Producto {
    /**
     * @param {string} id 
     * @param {string} titulo 
     * @param {number} precio 
     * @param {string} artista 
     * @param {number} anioLanzamiento 
     */
    constructor(id, titulo, precio, artista, anioLanzamiento) {
        super(id, titulo, precio);
        this.artista = artista;
        this.anioLanzamiento = anioLanzamiento;
        /**
         * Objeto que almacena las canciones del disco.
         * @type {Object.<number, string>}
         * Clave: número de pista, Valor: nombre de la canción
         */
        this.canciones = {};
    }

    /**
     * Añade una canción al disco.
     * @param {number} numeroPista Número de pista
     * @param {string} cancion Nombre de la canción
     */
    añadirCancion(numeroPista, cancion) {
        this.canciones[numeroPista] = cancion;
    }

    /**
     * Calcula la duración total estimada del disco.
     * Se asume que cada canción dura aproximadamente 3.5 minutos.
     * @return {number} Duración en minutos (estimación)
     */
    calcularDuracionTotal() {
        return Object.keys(this.canciones).length * 3.5;
    }

    // Getters
    getArtista() {
        return this.artista;
    }
    
    getAñoLanzamiento() {
        return this.añoLanzamiento;
    }
    
    getCanciones() {
        return {...this.canciones}; // Devuelve una copia del objeto
    }
}